packages/gatsby/src/internal-plugins/query-runner/query-compiler.js

```diff
@@ -217,6 +218,18 @@ class Runner {
             `${path.relative(store.getState().program.directory, filePath)}`
           )
       }
+
+      if (
+        query.isHook &&
+        process.env.NODE_ENV === `production` &&
+        typeof require(`react`).useContext !== `function`
+      ) {
+        report.panicOnBuild(
+          `You're likely using a version of React that doesn't support Hooks.
+Please update to React 16.8.0 or later to use the useStaticQuery hook.`
```

Comment:
you probably want to put in the same error message as above?
https://github.com/gatsbyjs/gatsby/pull/11588/files#diff-4bc662f92e2fed3fcf4fa02501ad6d3dR39

Reply:
Great catch! Fixed.